tests/auxil/bot_method_checks.py

```diff
@@ -92,10 +129,31 @@ def check_shortcut_signature(
                         f"For argument {kwarg} I expected {bot_sig.parameters[kwarg].annotation}, "
                         f"but got {shortcut_sig.parameters[kwarg].annotation}"
                     )
+            elif FORWARD_REF_PATTERN.search(str(shortcut_sig.parameters[kwarg])):
+                # If a shortcut signature contains a ForwardRef, the simple comparison of
+                # annotations can fail. Try and resolve the .__args__, then compare them.
+
+                for shortcut_arg, bot_arg in zip(
+                    shortcut_sig.parameters[kwarg].annotation.__args__,
+                    bot_sig.parameters[kwarg].annotation.__args__,
+                ):
+                    shortcut_arg_to_check = shortcut_arg  # for ruff
```

Comment:
why?

Reply:
ruff says I'm assigning something to a variable that is a part of the sequence I'm looping over, and that's bad apparently :)

Reply:
mh, kind of makes sense